proto/client-api/src/lib.rs

```diff
@@ -1,13 +1,57 @@
 //! Client API protocol implementation for Medea media server.
 
-// TODO: when using enum's Self will be in stable, remove it.
-#![allow(clippy::use_self)]
-
 use std::collections::HashMap;
 
+use macro_attr::*;
 use medea_macro::dispatchable;
+use newtype_derive::*;
 use serde::{de::Deserializer, ser::Serializer, Deserialize, Serialize};
 
+macro_attr! {
+    /// ID of [`Peer`].
+    #[cfg_attr(
+        feature = "medea",
+        derive(Deserialize, Debug, Hash, Eq, Default, PartialEq)
+    )]
+    #[cfg_attr(feature = "jason", derive(Serialize))]
+    #[derive(Clone, Copy, NewtypeDisplay!)]
+    pub struct PeerId(pub u64);
+}
+
+macro_attr! {
+    /// ID of [`MediaTrack`].
+    #[cfg_attr(
+        feature = "medea",
+        derive(Deserialize, Debug, Hash, Eq, Default, PartialEq)
+    )]
+    #[cfg_attr(feature = "jason", derive(Serialize))]
+    #[derive(Clone, Copy, NewtypeDisplay!)]
+    pub struct TrackId(pub u64);
+}
+
+/// Trait for providing function `increment()` which return current value + 1.
+#[cfg(feature = "medea")]
+pub trait Incrementable: Sized + Clone {
+    /// Returns current value + 1.
+    fn increment(&self) -> Self;
```

Comment:
It's better to name this method just `incr()`. This is a well-known and well-understandable shortening.